app/js/components/LabResultsList.jsx

```diff
@@ -0,0 +1,263 @@
+    const { dispatch } = this.props;
+    const { patientUUID } = this.state;
+    dispatch(constantsActions.getDateAndTimeFormat());
+    dispatch(patientAction.getPatient(patientUUID));
+    dispatch(patientAction.fetchPatientLabTestResults(patientUUID));
+  }
+
+  renderLabResultsTable(labResults) {
+    const { dateAndTimeFormat } = this.props;
+    const fields = ["TYPE", "STATUS", "REQUEST DATE", "SAMPLE DATE", "RESULT", "NORMAL RANGE"];
+
+    const columnMetadata = fields.map(columnName => ({
+      Header:
+  <span className={`labs-result-table-head-${columnName.replace(' ', '-').toLocaleLowerCase()}`}>
+    {columnName}
+  </span>,
+      accessor: "",
+      Cell: data => <Cell {...data} columnName={columnName} dateAndTimeFormat={dateAndTimeFormat} type="single" show={false} />,
+      className: `lab-results-list-cell-${columnName.replace(' ', '-').toLocaleLowerCase()}`,
+      headerClassName: `lab-result-list-header-${columnName.replace(' ', '-').toLocaleLowerCase()}`,
+    }));
+    return (
+      <div className="lab-results-list">
+        <SortableTable
+          data={labResults}
+          columnMetadata={columnMetadata}
+          filteredFields={fields}
+          filterType="none"
+          showFilter={false}
+          isSortable={false}
+          rowOnClick={this.handleShowResultsEntryPage}
+          noDataMessage="No orders found"
+          defaultPageSize={10}
+          subComponent={(row) => {
+            const isPanel = (row.original.order.concept.set) && (row.original.status !== "Ordered");
+            const rowFields = ["TYPE", "RESULT", "NORMAL RANGE"];
+            const rowColumnMetadata = rowFields.map(columnName => ({
+              accessor: "",
+              Cell: data => <Cell {...data} columnName={columnName} type="panel" />,
+              className: `lab-results-list-cell-${columnName.replace(' ', '-').toLocaleLowerCase()}`,
+              headerClassName: 'lab-results-list-header',
+            }));
+            if (isPanel) {
+              return (
+                <div className="collapsible-panel">
+                  <SortableTable
+                    data={row.original.encounter.obs[0].groupMembers}
+                    columnMetadata={rowColumnMetadata}
+                    collapseOnDataChange={false}
+                    collapseOnPageChange={false}
+                    showPagination={false}
+                    defaultPageSize={row.original.encounter.obs[0].groupMembers.length}
+                    defaultClassName=""
+                  />
+                </div>
+              );
+            }
+            return '';
+          }}
+        />
+      </div>
+    );
+  }
+
+  render() {
+    const { patients } = this.props;
+    const { patientUUID } = this.state;
+    const selectedPatient = patients[patientUUID] || {};
+    const { encounters = [], orders = [] } = selectedPatient;
+
+    const getPatientLabResults = () => {
+      const labResults = orders.map((order) => {
```

Comment:
@rhenshaw56  would it be possible to reverse this? Instead of starting with the orders, iterate over all the encounters, and then all the obs within the encounters.  Then filter the obs to only the ones you care about (datatype LabSet or Test).

Then for each obs you'd want:
matchedEncounter = the encounter that obs came from (if any)
matchedOrder = look up the matched order based on the Test Order Number on the matched encounter (if any)

Then I think you should have all the data you'd need to display the row.

Reply:
@mogoodrich yeah, that makes sense, especially for multiple encounters. Will do it right away and open a pr for it

Reply:
Great, thanks!